engine.py

```diff
@@ -611,6 +611,43 @@ def _set_project(self):
         self.logger.info("Setting Maya project to '%s'", proj_path)
         pm.mel.setProject(proj_path)
 
+    def get_session_path(self):
+        """
+        Returns the path to the current scene file if it resides on
+        disk. If unsaved, it returns an empty string.
```

Comment:
I think the comment is potentially misleading - it sounds to me like we are saying 'if there are unsaved changed, it returns None'. maybe just to be clear, it could say something like 'if the session has never been saved and isn't associated with a file on disk yet, `None` is returned.